nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Analyze/VersionFinder.cs

```diff
@@ -23,20 +23,46 @@ public static Task<VersionResult> GetVersionsAsync(
         return GetVersionsAsync(packageId, currentVersion, versionFilter, nugetContext, logger, cancellationToken);
     }
 
-    public static Task<VersionResult> GetVersionsAsync(
+    public static async Task<VersionResult> GetVersionsAsync(
         DependencyInfo dependencyInfo,
         NuGetContext nugetContext,
         ILogger logger,
         CancellationToken cancellationToken)
     {
-        var packageId = dependencyInfo.Name;
         var versionRange = VersionRange.Parse(dependencyInfo.Version);
         var currentVersion = versionRange.MinVersion!;
-        var versionFilter = CreateVersionFilter(dependencyInfo, versionRange);
+        var mergedResult = new VersionResult(currentVersion);
 
-        return GetVersionsAsync(packageId, currentVersion, versionFilter, nugetContext, logger, cancellationToken);
+        // Always split packageId by semicolons and trim each part
+        var packageIds = dependencyInfo.Name.Split(';').Select(id => id.Trim());
```

Comment:
I'd perfer to keep the old code and make sure that the `dependencyInfo` object that comes in has already been split.